pkg/api/v1/atlasdeployment_types.go

```diff
@@ -250,6 +250,25 @@ type AutoScalingSpec struct {
 	Compute *ComputeSpec `json:"compute,omitempty"`
 }
 
+// AdvancedAutoScalingSpec configures your deployment to automatically scale its storage
+type AdvancedAutoScalingSpec struct {
+	// Flag that indicates whether autopilot mode for Performance Advisor is enabled.
+	// The default is false.
+	AutoIndexingEnabled *bool `json:"autoIndexingEnabled,omitempty"`
```

Comment:
Found no mention of `autoIndexingEnabled` field in [the docs](https://www.mongodb.com/docs/atlas/reference/api/cluster-advanced/create-one-cluster-advanced/).
I think `autoScaling` has only `diskGB` and `compute` subfields.

This is not related to the issue but we can fix in bulk =)

Reply:
Done. Thanks)